pkg/globalmanager/controllers/lifelonglearning/upstream.go

```diff
@@ -98,6 +99,69 @@ func (c *Controller) appendStatusCondition(name, namespace string, cond sednav1.
 	})
 }
 
+func (c *Controller) updateStatusKnowledgeBase(name, namespace string, cd ConditionData) error {
+	client := c.client.LifelongLearningJobs(namespace)
+	return runtime.RetryUpdateStatus(name, namespace, func() error {
+		// check if models field exits
+		if cd.Output == nil || cd.Output.Models == nil || len(cd.Output.Models) == 0 {
+			klog.V(4).Infof("output models is nil, name is %s", name)
+			return nil
+		}
+
+		numberOfSamples := 0
+		aiModels := sednav1.AIModels{}
+		aiclasses := sednav1.AIClasses{}
+		aiclasses.ListOfAIClasses = cd.Output.Models[0].Classes
```

Comment:
It is recommended that slice lengths be allocated in advance.

Reply:
The following line has beed added to optimize slice allocation. 
`aiModels.ListOfAIModels = make([]sednav1.AIModel, 0, len(cd.Output.Models))`